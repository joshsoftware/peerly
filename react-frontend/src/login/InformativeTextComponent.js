import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const HrLine = styled.hr`
  width: 10%;
  background: ${({ theme }) =>
    theme === "dark" ? "var(--white)" : "var(--black)"};
`;

const Div = styled.div`
  color: ${({ theme }) => (theme === "dark" ? "var(--white)" : "var(--black)")};
  background-color: ${({ theme }) =>
    theme === "dark" ? "var(--black)" : "var(--white)"};
`;

const InformativeTextComponent = ({
  theme,
  className,
  informativeText,
  encouragementThought,
}) => (
  <Div theme={theme} className={className} data-testid="info">
    <h3>{informativeText}</h3>
    <HrLine theme={theme} data-testid="hrLine" />
    <h6>{encouragementThought}</h6>
  </Div>
);

InformativeTextComponent.defaultProps = {
  informativeText: "Lets Create the Office Positive",
  encouragementThought: "Encouragement Driven",
  theme: "dark",
};

InformativeTextComponent.propTypes = {
  className: PropTypes.string,
  informativeText: PropTypes.string.isRequired,
  encouragementThought: PropTypes.string.isRequired,
  theme: PropTypes.oneOf(["dark", "light"]),
};

export default React.memo(InformativeTextComponent);
