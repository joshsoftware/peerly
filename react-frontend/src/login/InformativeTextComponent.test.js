import React from "react";
import { render } from "@testing-library/react";
import InformativeTextComponent from "login/InformativeTextComponent";

describe("informative Text component test", () => {
  test("should render informative text component", () => {
    const { asFragment, getByTestId } = render(
      <InformativeTextComponent
        informativeText="Lets Create the Office Positive"
        encouragementThought="thought"
      />
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId("info")).toHaveStyleRule("color", "var(--white)");
    expect(getByTestId("info")).toHaveStyleRule(
      "background-color",
      "var(--black)"
    );
    expect(getByTestId("hrLine")).toHaveStyleRule("width", "10%");
    expect(getByTestId("hrLine")).toHaveStyleRule("background", "var(--white)");
  });

  test("test case for style", () => {
    const { getByTestId } = render(
      <InformativeTextComponent
        theme="light"
        informativeText="Lets Create the Office Positive"
        encouragementThought="thought"
      />
    );
    expect(getByTestId("info")).toHaveStyleRule("color", "var(--black)");
    expect(getByTestId("info")).toHaveStyleRule(
      "background-color",
      "var(--white)"
    );
    expect(getByTestId("hrLine")).toHaveStyleRule("background", "var(--black)");
    expect(getByTestId("hrLine")).toHaveStyleRule("width", "10%");
  });

  test("test case for classes", () => {
    const { getByTestId } = render(
      <InformativeTextComponent
        className="text-uppercase"
        informativeText="Lets Create the Office Positive"
        encouragementThought="thought"
      />
    );
    expect(getByTestId("info")).toHaveClass("text-uppercase");
  });
});
