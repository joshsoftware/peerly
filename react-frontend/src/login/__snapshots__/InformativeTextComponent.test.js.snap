// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`informative Text component test should render informative text component 1`] = `
<DocumentFragment>
  .c1 {
  width: 10%;
  background: var(--white);
}

.c0 {
  color: var(--white);
  background-color: var(--black);
}

<div
    class="c0 text-center undefined"
    data-testid="info"
  >
    <h3>
      Lets Create the Office Positive
    </h3>
    <hr
      class="c1"
      data-testid="hrLine"
    />
    <h6>
      thought
    </h6>
  </div>
</DocumentFragment>
`;
